chore(architecture diagrams): add C4 diagrams (#8)
NOAA-FIMS/ecosystemom: architecture_diagrams/system_context_diagram.md

# System context diagram

> This diagram presents the ecosystem simulation toolkit at the center, surrounded by its users and the external systems 

Reviewer: ```suggestion
This diagram presents the ecosystem-simulation toolkit at the center, surrounded by its users and the external systems 
```

it interacts with. The emphasis is on the people and the toolkit itself, rather than on technologies or low-level 
implementation details.

```mermaid
%%{init: { 'themeVariables': { 'clusterBkg': 'transparent' } } }%%
flowchart TB

%% Define styles for nodes
classDef borderless stroke-width:0px
classDef lightblue fill:#BBDEFB, color:#000
classDef lightgreen fill:#C8E6C9, color:#000
classDef lightpurple fill:#E1BEE7, color:#000

%% Set up legend for nodes
subgraph Legend
    direction LR
    Legend1[Person]:::lightblue
    Legend2[Toolkit Component]:::lightgreen
    Legend3[External System]:::lightpurple
end

%% Define users
subgraph "" 
    direction LR
    U1[<b>User 1</b><br/><br/>-<i>Person</i>-<br/><br/>Ecosystem Modeler]:::lightblue
    U2[<b>User 2</b><br/><br/>-<i>Person</i>-<br/><br/>General Data User]:::lightblue

>     U3[<b>User 3</b><br/><br/>-<i>Person</i>-<br/><br/>Stock Assessment Modeler]:::lightblue

Reviewer: I would prefer to take out the `-<i>Person<\i>` from each of these lines because it is clear from the legend that it is a "person" and it would allow for more room for the diagram. Same for `-<i>External System</i>-` on lines  43--35.

end

%% Define ecosystem simulation toolkit
subgraph "" 
    direction LR

>     EcosystemData[<b>ecosystemdata</b><br/><br/> -<i>R package</i>-<br/><br/> -Standardizes model output<br/>-Provides functions to connect with external tools to perform time series analysis]:::lightgreen

Reviewer: I do not quite understand what you are trying to imply with "Provides functions to connect with external tools to perform time series analysis"

>     EcosystemOM[<b>ecosystemom</b><br/><br/> -<i>R package</i>-<br/><br/> Samples observations from ecosystemdata for stock assessment testing]:::lightgreen

Reviewer: ```suggestion
    EcosystemOM[<b>ecosystemom</b><br/><br/> -<i>R package</i>-<br/><br/> Samples observations from ecosystemdata to fit to stock assessments]:::lightgreen
```

end

%% Define external systems
subgraph "" 
    direction TB

>     EM[<b>Ecosystem Models</b><br/><br/> -<i>External System</i>-<br/><br/> Simulate ecosystem dynamics and produce raw outputs]:::lightpurple

Reviewer: Can we list EWE and Atlantis here instead of "External System"?

>     DSEM[<b>Dynamic Structural Equation Models</b><br/><br/> -<i>External System</i>-<br/><br/> Fit dynamic structural equation models]:::lightpurple

Reviewer: Does mermaid have the ability to provide a hyper link to the DSEM repository with the text here?

>     SA[<b>Stock Assessment Models</b><br/><br/> -<i>External System</i>-<br/><br/> Analyze fisheries and environmental impacts on fish stocks]:::lightpurple

Reviewer: Can we list example models here like FIMS with a hyperlink instead of "External System"?
